android/app/src/main/java/app/organicmaps/location/LocationHelper.java

```diff
@@ -232,71 +231,44 @@ public void removeListener(@NonNull LocationListener listener)
     mListeners.remove(listener);
   }
 
-  private void calcLocationUpdatesInterval()
+  private long calcLocationUpdatesInterval()
   {
     if (RoutingController.get().isNavigating())
-    {
-      final @Framework.RouterType int router = Framework.nativeGetRouter();
-      switch (router)
-      {
-      case Framework.ROUTER_TYPE_PEDESTRIAN:
-        mInterval = INTERVAL_NAVIGATION_PEDESTRIAN_MS;
-        break;
-
-      case Framework.ROUTER_TYPE_VEHICLE:
-        mInterval = INTERVAL_NAVIGATION_VEHICLE_MS;
-        break;
-
-      case Framework.ROUTER_TYPE_BICYCLE:
-        mInterval = INTERVAL_NAVIGATION_BICYCLE_MS;
-        break;
-
-      case Framework.ROUTER_TYPE_TRANSIT:
-        // TODO: what is the interval should be for transit type?
-        mInterval = INTERVAL_NAVIGATION_PEDESTRIAN_MS;
-        break;
-
-      default:
-        throw new IllegalArgumentException("Unsupported router type: " + router);
-      }
-
-      Logger.d(TAG, "navigation = " + router + " interval = " + mInterval);
-      return;
-    }
+      return INTERVAL_NAVIGATION_MS;
 
-    int mode = LocationState.nativeGetMode();
+    final int mode = LocationState.nativeGetMode();
     switch (mode)
     {
       case LocationState.FOLLOW:
-        mInterval = INTERVAL_FOLLOW_MS;
-        break;
+        return INTERVAL_FOLLOW_MS;
       case LocationState.FOLLOW_AND_ROTATE:
-        mInterval = INTERVAL_FOLLOW_AND_ROTATE_MS;
-        break;
+        return INTERVAL_FOLLOW_AND_ROTATE_MS;
       case LocationState.NOT_FOLLOW:
       case LocationState.NOT_FOLLOW_NO_POSITION:
       case LocationState.PENDING_POSITION:
```

Comment:
Won't user receive location later when the delay is set to 3 seconds in case when location was not available?

Reply:
This is not an artificial delay, like sleep(3). We subscribe to location services with a given refresh interval. Implementation can take these 3 seconds into consideration to optimize battery usage.

Reply:
My question is about this situation:
1. A user wants to see his direction ASAP and presses the location button.
2. The system can return the location ASAP, but instead, it returns it ASAP + 3 seconds (or ASAP + 2 seconds).

Is it possible? If yes, then it's a bad UX because a user needs to wait for more time than in case of a 0 or 1-second delay.

Reply:
When the user presses the location button, the mode switches between FOLLOW and FOLLOW_AND_ROTATE modes, both of which use the fastest available refresh interval in this patch. NOT_FOLLOW mode is enabled then the user scrolls the map. In this case, precise location is not really important because the arrow is not in the center of the screen.

Reply:
For `PENDING_POSITION`  this PR changes `mInterval = INTERVAL_NOT_FOLLOW_MS;` to `return INTERVAL_NOT_FOLLOW_MS;`. That is it. What problem do you see here?

Reply:
INTERVAL_NOT_FOLLOW_MS = 3000;

I'm trying to understand if with interval = 0; the initial position will be reported to the user faster by 2 or 3 seconds or not.

In short: why not change interval for PENDING_POSITION to zero?

Reply:
> In short: why not change the interval for PENDING_POSITION to zero?

It is unclear to me why it was 3 seconds. I can change it to zero.

Reply:
I updated the code to have 0 refresh interval for PENDING_POSITION. Practical tests show that the first location is returned as soon as the device has it. In case of fused location, it almost instant. GPS can take a while to start up.